newIDE/app/src/VariablesList/index.js

```diff
@@ -237,27 +287,49 @@ export default class VariablesList extends React.Component<Props, State> {
 
   _renderEmpty() {
     return (
-      <div>
-        <EmptyMessage
-          style={styles.emptyExplanation}
-          messageStyle={styles.emptyExplanationMessage}
-        >
-          {this.props.emptyExplanationMessage || ''}
-        </EmptyMessage>
-        <EmptyMessage
-          style={styles.emptyExplanation}
-          messageStyle={styles.emptyExplanationMessage}
-        >
-          {this.props.emptyExplanationSecondMessage}
-        </EmptyMessage>
-      </div>
+      !!this.props.emptyExplanationMessage && (
+        <div>
+          <EmptyMessage
+            style={styles.emptyExplanation}
+            messageStyle={styles.emptyExplanationMessage}
+          >
+            {this.props.emptyExplanationMessage}
+          </EmptyMessage>
+          <EmptyMessage
+            style={styles.emptyExplanation}
+            messageStyle={styles.emptyExplanationMessage}
+          >
+            {this.props.emptyExplanationSecondMessage}
+          </EmptyMessage>
+        </div>
+      )
     );
   }
 
   render() {
-    const { variablesContainer } = this.props;
+    const { variablesContainer, inheritedVariablesContainer } = this.props;
     if (!variablesContainer) return null;
 
+    // map all object variables, if they have been passed
+    const containerObjectVariablesTree = inheritedVariablesContainer
+      ? mapFor(0, inheritedVariablesContainer.count(), index => {
+          const name = inheritedVariablesContainer.getNameAt(index);
+          if (!variablesContainer.has(name)) {
+            // Show only object variables that have no instance variable
+            const variable = inheritedVariablesContainer.getAt(index);
+            return this._renderVariableAndChildrenRows(
+              name,
+              variable,
+              0,
+              index,
+              undefined,
+              'parent'
+            );
+          }
+        })
+      : [];
+
+    // map all unique instance variables
```

Comment:
`// Display variables of the container`

(because the comment is actually false, these might be instance variables. These might be scene, global, object or instance variables)